app/services/suri_minter.rb

```diff
@@ -0,0 +1,56 @@
+require 'net/http'
+require 'uri'
+require 'errors'
+
+class SuriMinter < ActiveFedora::Noid::Minter::Base
+
+  protected
+
+  def next_id
+    retrieve_id
+  end
+
+  private
+
+  def valid?
+    druid.match?(/\A([a-z]{2})(\d{3})([a-z]{2})(\d{4})\z/i)
+  end
+
+  def retrieve_id
+    suri_request.basic_auth(Settings.suri.username, Settings.suri.password)
+    suri_connection.use_ssl = true
+    handle_response(suri_connection.request(suri_request))
+  rescue Net::OpenTimeout => err
+    raise Hydrox::SuriTimeoutError, "The connection to SURI has timed out."
+  end
+
+  def suri_connection
+    @suri_connection ||=  Net::HTTP.new(suri_uri.host, suri_uri.port)
+  end
+
+  def suri_uri
+    @suri_uri ||= URI.parse(Settings.suri.service_url)
+  end
+
+  def suri_request
+    @suri_request ||= Net::HTTP::Post.new(suri_uri.request_uri)
+  end
+
+  def handle_response(response)
+    case response.code
+    when "201"
+      druid(response.body)
+      return druid if valid?
+      raise Hydrox::SuriInvalidDruidReturned, "Invalid druid: #{druid}"
+    when "401"
+      raise Hydrox::SuriAuthenticationError, "Incorrect username/password provided to SURI."
+    else
+      raise Hydrox::SuriException, "An error occured communicating with SURI. #{response.code}: #{response.body}"
+    end
+  end
+
+  def druid(id=nil)
+    @druid ||= id.strip
```

Comment:
If you call this method for the first time without an argument (if a client calls `#valid?` before calling `#mint/#next_id`), this will fail with a `NoMethodError` because `nil` doesn't respond to `#strip`